stdlib/src/collections/list.mojo

```diff
@@ -493,16 +493,67 @@ struct List[T: CollectionElement, hint_trivial_type: Bool = False](
         self.capacity = new_capacity
 
     fn append(mut self, owned value: T):
-        """Appends a value to this list.
+        """Appends a value to this list. If there is no capacity left, resizes
+        to twice the current capacity. Except for 0 capacity where it sets 1.
 
         Args:
             value: The value to append.
         """
-        if self.size >= self.capacity:
-            self._realloc(max(1, self.capacity * 2))
-        (self.data + self.size).init_pointee_move(value^)
+        if len(self) >= self.capacity:
+            self._realloc(self.capacity * 2 + int(self.capacity == 0))
+        (self.data + len(self)).init_pointee_move(value^)
```

Comment:
**Suggestion** I suggest we keep this as `self.size` (and eventually `self._len`) instead of changing to `len(self)`. It's a minor difference, but it keeps the IR as simple as possible (a field access instead of a function call that will need to be inlined) and makes it clear to readers that no logic is being run to get the length. (Which isn't always true, e.g. `String.len` has logic to adjust the actual length by 1 to account for a null terminator.)